cmd/root.go

```diff
@@ -48,35 +48,40 @@ of Cloud Storage FUSE, see https://cloud.google.com/storage/docs/gcs-fuse.`,
 			return mountFn(configObj)
 		},
 	}
-	cobra.OnInitialize(initConfig)
+	v := viper.New()
+	cobra.OnInitialize(initConfig(v))
 	rootCmd.PersistentFlags().StringVar(&cfgFile, "config-file", "", "The path to the config file where all gcsfuse related config needs to be specified. "+
 		"Refer to 'https://cloud.google.com/storage/docs/gcsfuse-cli#config-file' for possible configurations.")
 
 	// Add all the other flags.
-	if err := cfg.BindFlags(rootCmd.PersistentFlags()); err != nil {
+	if err := cfg.BindFlags(v, rootCmd.PersistentFlags()); err != nil {
 		return nil, fmt.Errorf("error while declaring/binding flags: %w", err)
 	}
 	return rootCmd, nil
 }
 
-func initConfig() {
-	if cfgFile != "" {
-		cfgFile, err := util.GetResolvedPath(cfgFile)
-		if err != nil {
-			logger.Fatal("error while resolving config-file path[%s]: %v", cfgFile, err)
-		}
-		viper.SetConfigFile(cfgFile)
-		viper.SetConfigType("yaml")
-		if err := viper.ReadInConfig(); err != nil {
-			logger.Fatal("error while reading the config: %v", err)
+func initConfig(v *viper.Viper) func() {
+	// Returning a function instead of using a package-level variable since it
+	// allows for the viper instance to be GC'ed.
```

Comment:
What is GC'ed? Garbage collected?

Reply:
Yes

Reply:
Raised https://github.com/GoogleCloudPlatform/gcsfuse/pull/2079 for the fix.